wallet/src/plan.rs

```diff
@@ -264,6 +268,272 @@ where
     Ok(plan)
 }
 
+#[allow(clippy::too_many_arguments)]
+#[allow(dead_code)]
+#[instrument(skip(fvk, view, rng, swap_nft_note, fee, source_address))]
+pub async fn swap_claim<V, R>(
+    fvk: &FullViewingKey,
+    view: &mut V,
+    mut rng: R,
+    swap_nft_note: Note,
+    fee: u64,
+    source_address: Option<u64>,
+) -> Result<TransactionPlan, anyhow::Error>
+where
+    V: ViewClient,
+    R: RngCore + CryptoRng,
+{
+    tracing::debug!(?swap_nft_note, ?fee, ?source_address);
+
+    return Err(anyhow::anyhow!("not implemented"));
+
+    let chain_params = view.chain_params().await?;
+
+    // TODO: need to fetch clearing price (`BatchSwapOutputData`) for the
+    // swap.
+
+    let mut plan = TransactionPlan {
+        chain_id: chain_params.chain_id,
+        fee: Fee(fee),
+        ..Default::default()
+    };
+
+    // Add a `SwapClaimPlan` action:
+    // plan.actions.push(
+    //     SwapClaimPlan::new(
+    //         &mut rng,
+    //         swap_nft_note,
+    //         swap_nft_position,
+    //         // The `fvk` is always the claim address.
+    //         fvk.address(),
+    //         fee,
+    //         output_data,
+    //         anchor,
+    //         trading_pair,
+    //     )
+    //     .into(),
+    // );
+
+    // The value we need to spend is 1 unit of the swap NFT, plus fees.
+    let mut value_to_spend: HashMap<Denom, u64> = HashMap::new();
+    // *value_to_spend.entry(swap_nft_note.denom()).or_default() += input_value;
```

Comment:
Commented out because `denom()` on the NFT note is not a thing yet?

Reply:
Commented out because it wasn't clear how to implement `SwapClaim` here just yet